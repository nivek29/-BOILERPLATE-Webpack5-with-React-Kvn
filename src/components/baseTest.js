
const nombre   = 'Fernando';
const apellido = 'Herrera';


// const nombreCompleto = nombre + ' ' + apellidosss
const nombreCompleto = `${ nombre } ${ apellido }`;

// console.log( nombreCompleto );


export function getSaludo( nombre = 'Carlos' ) {
    return 'Hola ' + nombre + '!';
}