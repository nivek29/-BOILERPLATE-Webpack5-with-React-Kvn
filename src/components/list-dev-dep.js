import React from "react";
import ReactDOM from "react-dom";
import "antd/dist/antd.css";

import { Table, Tag, Space } from "antd";

export const ListDev = () => {
	const columns = [
		{
			title: "Name",
			dataIndex: "name",
			key: "name",
		},
		{
			title: "Version",
			key: "tags",
			dataIndex: "tags",
			// eslint-disable-next-line react/display-name
			render: (tags) => (
				<>
					{tags.map((tag) => {
						return (
							<Tag color="green" key={tag}>
								{tag.toUpperCase()}
							</Tag>
						);
					})}
				</>
			),
		},
	];
	const data = [
		{
			key: "1",
			name: "@babel/core",

			tags: ["7.13.15"],
		},
		{
			key: "2",
			name: "@babel/preset-env",

			tags: ["7.13.15"],
		},
		{
			key: "3",
			name: "@babel/preset-react",

			tags: ["7.13.13"],
		},
		{
			key: "4",
			name: "@pmmmwh/react-refresh-webpack-plugin",

			tags: ["0.4.3"],
		},
		{
			key: "5",
			name: "@types/jest",

			tags: ["26.0.23"],
		},
		{
			key: "6",
			name: "@wojtekmaj/enzyme-adapter-react-17",

			tags: ["0.6.1"],
		},
		{
			key: "7",
			name: "babel-eslint",

			tags: ["10.1.0"],
		},
		{
			key: "8",
			name: "babel-jest",

			tags: ["26.6.3"],
		},
		{
			key: "9",
			name: "babel-loader",

			tags: ["8.2.2"],
		},
		{
			key: "10",
			name: "bc-console",

			tags: ["0.0.2"],
		},
		{
			key: "11",
			name: "clean-webpack-plugin",

			tags: ["3.0.0"],
		},
		{
			key: "12",
			name: "css-loader",

			tags: ["5.2.4"],
		},
		{
			key: "13",
			name: "enzyme",

			tags: ["3.11.0"],
		},
		{
			key: "14",
			name: "eslint",

			tags: ["7.25.0"],
		},
		{
			key: "15",
			name: "eslint-loader",

			tags: ["4.0.2"],
		},
		{
			key: "16",
			name: "eslint-plugin-react",

			tags: ["7.23.2"],
		},
		{
			key: "17",
			name: "eslint-watch",

			tags: ["7.0.0"],
		},
		{
			key: "18",
			name: "file-loader",

			tags: ["6.2.0"],
		},
		{
			key: "19",
			name: "jest",

			tags: ["26.6.3"],
		},
		{
			key: "20",
			name: "jest-enzyme",

			tags: ["7.1.2"],
		},
		{
			key: "21",
			name: "mini-css-extract-plugin",

			tags: ["1.5.0"],
		},
		{
			key: "22",
			name: "prettier",

			tags: ["2.2.1"],
		},
		{
			key: "23",
			name: "prettier-webpack-plugin",

			tags: ["1.2.0"],
		},
		{
			key: "24",
			name: "react-query",

			tags: ["3.13.11"],
		},
		{
			key: "25",
			name: "react-refresh",

			tags: ["0.10.0"],
		},
		{
			key: "26",
			name: "sass",

			tags: ["1.32.11"],
		},
		{
			key: "27",
			name: "sass-loader",

			tags: ["11.0.1"],
		},
		{
			key: "28",
			name: "style-loader",

			tags: ["2.0.0"],
		},
		{
			key: "29",
			name: "webpack",

			tags: ["5.34.0"],
		},
		{
			key: "30",
			name: "webpack-cli",

			tags: ["4.6.0"],
		},
		{
			key: "31",
			name: "webpack-dev-server",

			tags: ["3.11.2"],
		},
		{
			key: "32",
			name: "webpack-merge",

			tags: ["5.7.3"],
		},
		{
			key: "33",
			name: "html-webpack-plugin",

			tags: ["5.3.1"],
		},
	];

	return <Table columns={columns} dataSource={data} />;
};
