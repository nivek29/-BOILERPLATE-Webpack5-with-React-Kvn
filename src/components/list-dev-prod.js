import React from "react";
import ReactDOM from "react-dom";
import "antd/dist/antd.css";

import { Table, Tag, Space } from "antd";

export const ListProd = () => {
	const columns = [
		{
			title: "Name",
			dataIndex: "name",
			key: "name",
		},
		{
			title: "Version",
			key: "tags",
			dataIndex: "tags",
			// eslint-disable-next-line react/display-name
			render: (tags) => (
				<>
					{tags.map((tag) => {
						return (
							<Tag color="green" key={tag}>
								{tag.toUpperCase()}
							</Tag>
						);
					})}
				</>
			),
		},
	];
	const data = [
		{
			key: "1",
			name: "@ant-design/icons",

			tags: ["4.6.2"],
		},
		{
			key: "2",
			name: "@material-ui/core",

			tags: ["4.11.4"],
		},
		{
			key: "3",
			name: "antd",

			tags: ["4.15.2"],
		},
		{
			key: "4",
			name: "core-js",

			tags: ["3.10.2"],
		},
		{
			key: "5",
			name: "prop-types",

			tags: ["15.7.2"],
		},
		{
			key: "6",
			name: "react",

			tags: ["17.0.2"],
		},
		{
			key: "7",
			name: "react-dom",

			tags: ["17.0.2"],
		},
		{
			key: "8",
			name: "react-router-dom",

			tags: ["5.2.0"],
		},
		{
			key: "9",
			name: "redux-saga",

			tags: ["1.1.3"],
		},
	];

	return <Table columns={columns} dataSource={data} />;
};
