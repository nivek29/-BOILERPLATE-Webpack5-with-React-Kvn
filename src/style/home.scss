
body {
  margin:0;
  padding:0;
  background-color: rgb(48, 47, 47) !important;
  height: 0px !important
}
.home{
  margin-top: 11em;
  display: flex;
  justify-content: center;
}
.btn-home{
  margin-top: 3em;
  display: flex;
  justify-content: center;
}
img{
  width: 50%;
}
button{
  margin-left: 10px !important;
  color: rgb(48, 47, 47) !important;
  border: #00D8FF !important;
  background-color: #00D8FF !important;
  font-weight: bold !important;
}