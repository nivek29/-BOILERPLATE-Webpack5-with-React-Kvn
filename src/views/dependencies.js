import { React, Fragment } from "react";
import "../style/dependencies.scss";
import { ListDev } from "../components/list-dev-dep";
import { ListProd } from "../components/list-dev-prod";

export const Dependencies = () => {
	return (
		<Fragment>
			<div className="list-dependencies">
				<p className="parr-dependencies">
					React Boiler-Plate custom , contains the necessary tools for
					development with React. In addition to a set of extra tools
					for unit tests and design components.
				</p>
			</div>
			<div className="list-dependencies-dev">
				<p className="parr-dependencies-dev">Develop dependencies</p>
				<ListDev></ListDev>
			</div>
			<div className="list-dependencies-prod">
				<p className="parr-dependencies-prod">
					Production dependencies
				</p>
				<ListProd></ListProd>
			</div>
		</Fragment>
	);
};
