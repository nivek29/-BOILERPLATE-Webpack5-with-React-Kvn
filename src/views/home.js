import React, { Fragment } from "react";
import { Link } from "react-router-dom";
import "../style/home.scss";
import "antd/dist/antd.css";
import photoReact from "../img/Daco_6126558.png";
import { Button } from "antd";
//import { Antd } from "../components/antd";

export const Home = () => {
	return (
		<Fragment>
			<div className="home">
				<img src={photoReact}></img>
			</div>
			<div className="btn-home">
				<Link to="/dependencies">
					<Button type="primary" shape="round" size="large">
						Dependencies
					</Button>
				</Link>

				<Button type="primary" shape="round" size="large">
					Examples
				</Button>
				<Button type="primary" shape="round" size="large">
					NpmÂ´s
				</Button>
			</div>
		</Fragment>
	);
};
