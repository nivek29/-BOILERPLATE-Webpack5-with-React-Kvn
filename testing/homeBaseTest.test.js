/* eslint-disable no-undef */

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { getSaludo } from '../src/components/baseTest';
import { Home } from '../src/views/home';

describe('Pruebas en home', () => {

    test('getSaludo debe de retornar Hola Fernando!', () => {
        
        const nombre = 'Fernando';

        const saludo = getSaludo( nombre );
        //jest permite utilizar los metodos de la clase 

        expect( saludo ).toBe( 'Hola ' + nombre + '!');
        //toBe compara que el resultado de una variable sea el resultado que estamo esperando

    });
    
    // getSaludo debe de retorar Hola Carlos! si no hay argumento nombre
    test('getSaludo debe de retorar Hola Carlos! si no hay argumento nombre', () => {
    
        const saludo = getSaludo();
        expect( saludo ).toBe( 'Hola Carlos!' );

    });
    
    test(' se esta renderizando el componente home', () => {
        // eslint-disable-next-line react/react-in-jsx-scope
        const wrapper = shallow( <Home></Home> );
        expect(wrapper).toMatchSnapshot();
    }); 
    

    
});