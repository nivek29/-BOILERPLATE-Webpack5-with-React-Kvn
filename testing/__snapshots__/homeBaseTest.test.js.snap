// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en home  se esta renderizando el componente home 1`] = `
<Fragment>
  <div
    className="home"
  >
    <img
      src={Object {}}
    />
  </div>
  <div
    className="btn-home"
  >
    <Link
      to="/dependencies"
    >
      <Button
        shape="round"
        size="large"
        type="primary"
      >
        Dependencies
      </Button>
    </Link>
    <Button
      shape="round"
      size="large"
      type="primary"
    >
      Examples
    </Button>
    <Button
      shape="round"
      size="large"
      type="primary"
    >
      NpmÂ´s
    </Button>
  </div>
</Fragment>
`;
